src/utilities/createRouterReject.ts

```diff
@@ -50,26 +57,29 @@ export function createRouterReject({
 
   const getRejectionRoute: GetRejectionRoute = (type) => {
     const component = markRaw(getRejectionComponent(type))
+
     const route = {
       name: type,
       path: '',
       component,
     }
 
-    const resolved = createResolvedRoute({
+    const resolved = {
       matched: route,
       matches: [route],
       name: type,
-      path: '',
-      query: '',
+      query: createRouterRouteQuery(''),
       params: {},
-      depth: 0,
-      isRejection: true,
-    })
+      [isRejectionRouteSymbol]: true,
+    }
 
     return resolved
   }
 
+  const isRejectionRoute: IsRejectionRoute = (route: RouterRejectionRoute) => {
+    return route[isRejectionRouteSymbol] === true
+  }
+
```

Comment:
I thought this turned out nicely. The type for a rejection route is slightly wider than RouterRoute. But since the `isRejectionRouteSymbol` is optional it accepts `RouterRoute` just fine.

Reply:
yes! Love that ❤️